### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by dylanvaughn**
@nlamprian I updated the Docker stuff to include the `make teleop`, which assumes that `make bringup` is already running.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+RUN wget http://packages.osrfoundation.org/gazebo.key
+RUN apt-key add gazebo.key
+RUN apt-get update && apt-get install -y     \
+        python3-apt                          \
+        python3-colcon-common-extensions     \
+        python3-pip                          \
+        python3-vcstool                      \
+        ros-noetic-desktop-full              \
+        ros-noetic-gazebo-plugins            \
```
Comment (nlamprian) [inline Dockerfile]: Remove gazebo-plugins and add gazebo-ros-control

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        ros-noetic-joint-state-publisher-gui \
```
Comment (nlamprian) [inline Dockerfile]: You can remove this

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        ros-noetic-key-teleop                \
```
Comment (nlamprian) [inline Dockerfile]: You could change this to ros-noetic-teleop-tools

File: software/.catkin_workspace [hunk 4]
```diff
@@ -0,0 +1 @@
+# This file currently only serves to mark the location of a catkin workspace for tool integration
```
Comment (nlamprian) [inline software/.catkin_workspace]: Remove and ignore this file as well

File: docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -e
+
+# setup ros environment
+source "/opt/ros/noetic/setup.bash"
+if [ -f "/code/software/devel/setup.bash" ]; then
```
Comment (nlamprian) [inline docker-entrypoint.sh]: If you'd like to use `colcon build` in the compile command, change `devel` to `install` here

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,23 @@
+GUI_OPTIONS=--env=DISPLAY --env=QT_X11_NO_MITSHM=1 -v /tmp/.X11-unix:/tmp/.X11-unix
+GENERAL_OPTIONS=-it --rm --privileged --net=host --volume $(PWD)/software:/software
```
Comment (nlamprian) [inline Makefile:2]: Maybe you'd like to assign the volume to a more representative name, like catkin_ws or workspace. Up to you.

**Review by nlamprian — APPROVED** [review]

File: docker-entrypoint.sh (hunk 5 above)
Comment (dylanvaughn) [reply docker-entrypoint.sh]: I tried to switch this to `colcon` but ran into issues - I'll save that for a different PR

**Pull request merged**